### DISCUSSION TITLE
Different icons for different filetypes

### DISCUSSION THREAD

**Discussion by goerz** [hedyhli/outline.nvim#78] [Help] [answered]
Is it possible to set a different `icons` table (in `symbols`) for different filetypes? For example, I would like to use a full set of symbols for code outlines, but no symbols for markdown files.

**Comment by hedyhli** [comment, marked as answer]
Hi!

Yes, you can. There's a [section in the docs](https://github.com/hedyhli/outline.nvim#custom-icons) dedicated to customizing this specifically.

In fact, I use this myself:

https://github.com/hedyhli/dotfiles/blob/8b41a2c8135cd579ac7adacb3ca94de993a6b6be/.config/nvim/lua/plugins/outline.lua#L77

```lua
    symbols = {
      icon_fetcher = function(k)
        local ft = vim.api.nvim_buf_get_option(require('outline').current.code.buf, "ft")
        -- There can only be kind String in markdown so... let's not have the
        -- eye candy here
        if ft == 'markdown' and k == 'String' then
          return ""
        end
        return false
      end,
      icon_source = "lspkind",
    }
```

No icons are shown when it's markdown. Falls back to `lspkind` otherwise in my case.

**Reply by goerz** [reply]
That works perfectly, thanks! I had seen the section in the docs, but I didn't think of getting `ft` the way you do in that snippet. That might be worth adding to the Tips in the README.

**Reply by hedyhli** [reply]
You're right! I hadn't included this specific example previously as it was bit of a hack, using an undocumented API. I've now added this feature officially -- `icon_fetcher` can now accept [a second argument](https://github.com/hedyhli/outline.nvim?tab=readme-ov-file#disable-icons-for-a-specific-filetype) which will be the code buffer number.

The snippet you're using above will still continue to work.
